[azumieflare] iP

azumieflare opened the pull request:
# Duke++

- **Easy** to use
- **Saves** persists across sessions
- ~~Probably~~ Will be improved on


## How to use
1. Download the .jar
2. Run the command java -jar (.jar name)
3. Watch the magic happen

## Quick peek
```java
public static void main(String[] args) {
    new Duke("./tasks.txt").run();
}
```

## Future additions
- [x] Run without crashing :grin:
- [ ] Find and fix bugs
- [ ] Improve user experience

> [!NOTE]
> Expect more improvements in the coming [future](https://www.youtube.com/watch?v=dQw4w9WgXcQ)!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

azumieflare committed:
Level-0

azumieflare committed:
Level-1

azumieflare committed:
Level-2

azumieflare committed:
Level-3

azumieflare committed:
Level-4

azumieflare committed:
Level-4

azumieflare committed:
Level-5

azumieflare committed:
Level-6

azumieflare committed:
Changed to enum

azumieflare committed:
Changed to enum

azumieflare committed:
Added missing tags

azumieflare committed:
Add new features - save and load.

azumieflare committed:
remove unneeded echo method

azumieflare committed:
Merge branch 'master' into branch-Level-7

azumieflare committed:
Merge branch 'branch-Level-7'

azumieflare committed:
Changed input format for deadline and event from String to LocalDate.

azumieflare committed:
-

azumieflare committed:
Revert "-"

This reverts commit f5002a0094fedd13db1299baa31d480fd8bdcbaf.

azumieflare committed:
Merge branch 'branch-level-8'

azumieflare committed:
Minor fix to save and loading

azumieflare committed:
Split the main method to include more OOP by adding the classes Storage, TaskList, Ui and Parser.

azumieflare committed:
Merge branch 'branch-A-MoreOOP'

azumieflare committed:
Package all related files into a common package duke.

azumieflare committed:
Merge branch 'add-gradle-support' of https://github.com/azumieflare/ip

azumieflare committed:
Add gradle support, TaskListTest.java and ToDoTest.java for testing. Also fixed to follow coding standards and added JavaDocs.

azumieflare committed:
Add find

azumieflare committed:
Add find

azumieflare committed:
Add CodingStandard

azumieflare committed:
Add JavaDoc

azumieflare committed:
Merge branch 'branch-A-CodingStandard'

azumieflare committed:
Merge branch 'branch-A-JavaDoc'

azumieflare committed:
Merge branch 'branch-Level-9'

azumieflare committed:
Add Checkstyle

azumieflare committed:
Add GUI integration

azumieflare committed:
Merge branch 'branch-Level-10'

azumieflare committed:
Change how the parser works

azumieflare committed:
Merge branch 'branch-A-Varargs'

billyhoce commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,176 @@
+                }
+
+            } catch (DukeException de) {
+                String text = "\t____________________________________________________________\n"
```
Maybe naming this "output" will allow for easier understanding!

billyhoce commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,176 @@
+        } catch (DateTimeParseException dtpe) {
+            return "Please enter a datetime format of yyyy-mm-dd.\n";
+        }
+        // return "Duke heard: " + input;
```
I'm not sure what this commented-out line is for! If it is not needed, you could delete it, or explain its purpose!

billyhoce commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,176 @@
+        try {
+            stringList = parser.parse(input);
+            Options choice = optionType(stringList.get(0));
+
```
I think this empty line can be removed

billyhoce commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+        LocalDate start;
+        LocalDate finish;
+        int index;
+        Task t;
```
Task variable t is used throughout this class, would it be better if it was named "task" instead for better clarity?

billyhoce commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+/**
+ * TaskList to handle all operations related to tasks.
+ */
+
```
Might want to remove this line!

billyhoce commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,173 @@
+        trail.remove(0);
+
+        switch (choice) {
+        case bye: // TODO make this exit the chatbot
```
Good job following the case indentation standard of this course!

billyhoce reviewed (commented):
Looks good overall! Well commented and documented with Javadoc. Code is neat with just some slight formatting fixes to be made

azumieflare committed:
Update code quality

Made some changes to code quality based off feedback from PR reviewers.

Also reduced redundant code by reusing existing methods.

azumieflare committed:
Add assertions

Added some assertions in Parser.java and TaskList.java to ensure that bugs do not occur.

azumieflare committed:
Merge pull request #2 from azumieflare/branch-A-Assertions

Add assertions

azumieflare committed:
Merge pull request #1 from azumieflare/branch-A-CodeQuality

Update code quality

azumieflare committed:
Add CI

azumieflare committed:
Add BCD-Extension(C-Update and C-Help)

C-Help was done in previous commits.

C-Update is done by adding a new command update that allows tasks to update its name and (by/to/from) fields.

This is done by the command: update (taskname) (name/by/to/from) (update value).

azumieflare committed:
Merge branch 'branch-BCD-Extension'

alex-setyawan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Deadline task.
+ */
+public class Deadline extends Task {
```
Perhaps you might want to change your class name from Deadline to DeadlineTask to better follow naming standards?

alex-setyawan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,36 @@
+    public String getDescription() {
+        return this.description;
+    }
+    public void updateDescription(String description) {
```
Maybe this can be DescriptionUpdater, or DescUpdater? Using nouns for descriptions provide greater clarity about the purpose of functions!

azumieflare committed:
Improve GUI

The .fxml files were changed to include text wrapping and
better fit to default resolution.

Additionally added a default message to GUI when launched.

azumieflare committed:
Add personality to chatbot

Rename the whole chatbot from existing name Duke to new name Teemo.

Also add personality to custom error messages.

azumieflare committed:
Merge branch 'branch-A-Personality'

azumieflare committed:
Add more test cases for TaskList.java and Parser.java

The tests should guarantee that both are working as intended.

Also fixed some javadocs standards.

azumieflare committed:
Remove CLI aspects

Remove existing CLI components of the chatbot to make the overall
code cleaner.

azumieflare committed:
Add Ui.png

azumieflare committed:
Add Ui.png

azumieflare committed:
Add user guide

azumieflare committed:
Fix minor bugs for creating .jar

azumieflare committed:
Fix minor bugs
